src/store/index.ts

```diff
@@ -80,41 +90,103 @@ export default createStore<State>({
         payload: messageData.ciphertext
       });
     },
-    async uploadFile({ dispatch }, request: UploadRequest) {
+    async uploadFile({ dispatch, getters, state }, request: UploadRequest) {
```

Comment:
It's probably worth making all the action return types explicit, to avoid accidental returns.

```suggestion
    async uploadFile(
      { dispatch, getters, state },
      request: UploadRequest
    ): Promise<void> {
```